core/src/main/java/org/infinispan/security/GlobalSecurityManager.java

```diff
@@ -16,11 +18,12 @@
    /**
     * Returns the global ACL cache
     */
-   Cache<?, ?> globalACLCache();
+   <K, V> Map<K, V> globalACLCache();
 
    /**
     * Flushes the ACL cache for this node
+    * @return
     */
-   void flushGlobalACLCache();
+   CompletionStage<Void> flushGlobalACLCache();
```

Comment:
This is public API, so we should deprecate rather than change it?

Reply:
Not really: the only way to obtain a GlobalSecurityManager is through the component registry. I wouldn't worry about it